api/controllers/MajorController.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Extends REST handling with custom logic for the major model. 
+ * @implements Controller
+ * @module
+ */
+
+const RestController = require("./RestController");
+
+module.exports = {
+
+    /**
+     * Handles request to update or create a major data record using form data.
+     * @argument {external:Request} request -  The HTTP request.
+     * @argument {external:Response} response - The HTTP response.
+     * @public
+     * @async
+     */
+    majorFormSubmitted: async function (request, response) {
```

Comment:
Generally, controller method names should match the method names in `RestController`. It's not technically true with the sails approach, but we are pretending that most controllers extend `RestController` and override its methods. So, the model name should not appear in the method names. 
Judging from the comment, this method is not exactly `createFormSubmitted` or `editFormSubmitted`, since it apparently handles both actions? Let's discuss the naming convention for that situation.